### ISSUE TITLE
lx brand alpine cp/cat (busybox) duplicates data

### ISSUE THREAD

**Issue by h0lzi** [joyent/illumos-joyent#122]
joyent_20160622T234741Z:

```
00156c80-5535-484d-b941-9e63e06bed6e:~# cat /etc/motd 
   __        .                   .
 _|  |_      | .-. .  . .-. :--. |-
|_    _|     ;|   ||  |(.-' |  | |
  |__|   `--'  `-' `;-| `-' '  ' `-'
                   /  ;  Instance (Alpine Linux 3.4 20160620)
                   `-'   https://docs.joyent.com/images/container-native-linux

00156c80-5535-484d-b941-9e63e06bed6e:~# cat test.txt
test
00156c80-5535-484d-b941-9e63e06bed6e:~# cp test.txt test2.txt
00156c80-5535-484d-b941-9e63e06bed6e:~# cat test2.txt 
test
test

00156c80-5535-484d-b941-9e63e06bed6e:~# cat test.txt >  out.txt
00156c80-5535-484d-b941-9e63e06bed6e:~# cat out.txt 
test
test
```

/native/usr/bin/cp is fine.

**Comment by h0lzi** [comment]
strace cp test.txt test2.txt
execve("/bin/cp", ["cp", "test.txt", "test2.txt"], [/* 16 vars */]) = 0
arch_prctl(ARCH_SET_FS, 0x7ffffee8ac28) = 0
set_tid_address(0x7ffffee8ac60)         = 8950
mprotect(0x7ffffee87000, 4096, PROT_READ) = 0
mprotect(0x7fffff2c0000, 16384, PROT_READ) = 0
getuid()                                = 0
stat("test.txt", {st_mode=S_IFREG|0644, st_size=5, ...}) = 0
stat("test2.txt", {st_mode=S_IFREG|0644, st_size=10, ...}) = 0
stat("test.txt", {st_mode=S_IFREG|0644, st_size=5, ...}) = 0
lstat("test2.txt", {st_mode=S_IFREG|0644, st_size=10, ...}) = 0
open("test.txt", O_RDONLY)              = 3
open("test2.txt", O_WRONLY|O_CREAT|O_EXCL, 0100644) = -1 EEXIST (File exists)
unlink("test2.txt")                     = 0
open("test2.txt", O_WRONLY|O_CREAT|O_EXCL, 0100644) = 4
sendfile(4, 3, NULL, 16777216)          = -1 EINVAL (Invalid argument)
mmap(NULL, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7ffffefb0000
read(3, "test\n", 16384)                = 5
write(4, "test\n", 5)                   = 5
read(3, "", 16384)                      = 0
munmap(0x7ffffefb0000, 16384)           = 0
close(4)                                = 0
close(3)                                = 0
exit_group(0)                           = ?
+++ exited with 0 +++

**Comment by jjelinek** [comment]
Thanks for reporting this, I filed OS-5498 for this bug.

**Comment by pfmooney** [comment]
This should be fixed in 6d2db2a21cd0c3de9f0fa59e794736c6c168a322

Thanks
